MdeModulePkg/Bus/Pci/NvmExpressDxe/UnitTest/MediaSanitizeUnitTest.c

```diff
@@ -0,0 +1,1208 @@
+/** @file -- MediaSanitizeUnitTest.c
+  Placeholder/framework for developing a Media Sanitize unit test package.
+
+  Copyright (c) Microsoft Corporation.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <Uefi.h>
+#include <Library/BaseLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <Library/UefiLib.h>
+#include <Library/DebugLib.h>
+#include <Library/UnitTestLib.h>
+#include <Library/UefiBootServicesTableLib.h>
+#include <Library/UefiApplicationEntryPoint.h>
+#include <Library/UefiRuntimeServicesTableLib.h>
+#include <Uefi/UefiBaseType.h>
+#include <Library/UefiBootManagerLib.h>
+#include <Protocol/NvmExpressPassthru.h>
+
+#include "../NvmExpress.h"
+#include "../NvmExpressBlockIo.h"
+#include "../NvmExpressMediaSanitize.h"
+#include "../NvmExpressHci.h"
+#include "../../../Include/Protocol/MediaSanitize.h"
+
+// #pragma warning ( disable : 4100 )
+
+EFI_STATUS
+EFIAPI
+NvmeDeviceUnitTestPassthru (
+  IN     EFI_NVM_EXPRESS_PASS_THRU_PROTOCOL        *This,
+  IN     UINT32                                    NamespaceId,
+  IN OUT EFI_NVM_EXPRESS_PASS_THRU_COMMAND_PACKET  *Packet,
+  IN     EFI_EVENT                                 Event OPTIONAL
+  )
+{
+  //
+  // Parse command packet for unit testing
+  //
+  // NVME_DEVICE_PRIVATE_DATA                  *Device;
+  EFI_NVM_EXPRESS_COMMAND     *Command;
+  EFI_NVM_EXPRESS_COMPLETION  *Completion;
+  EFI_STATUS                  Status;
+  NVME_CQ                     *Cqe;
+
+  DEBUG ((DEBUG_VERBOSE, "[          ] NvmeDeviceUnitTestPassthru: Enter\n"));

```

Comment:
Good question. No, they are not. It was useful as I was developing the unit tests and getting an understanding of the flow of how the host application was invoked. I will remove the enter/exit debug print statements.

Reply:
adding to it.  Can you help explain what the "[         ]" part of the debug message is for.    Also there is a UT_LOG_* macro that will do extra logging for the unit tests and get it added to the report.

Reply:
The extra "[  ]" was to make the output look similar to the other UT outputs. I can convert all the debug prints to use the UT LOG macro if that is preferred. I was unaware of the logging macro when I first wrote the file... this would be preferred for me as well.